### PR TITLE
mavlink_mission: support missions with int params

### PR DESCRIPTION
**Pull request by julianoes**
This brings support for the messages MISSION_REQUEST_INT and
MISSION_ITEM_INT which raises the lat/lon accuracy for waypoints.

Tested in SITL.

### CODE REVIEW

**Comment by LorenzMeier** [comment]
LGTM

**Comment by julianoes** [comment]
@LorenzMeier what's the state here? Can you please have a quick look if it's safe? If not, I'll change it to avoid the pointer casting.

**Comment by julianoes** [comment]
Rebased and improved the comments.

**Comment by dagar** [comment]
The reinterpret_cast makes me a little sad, but I don't see a simple alternative. 
Do we actually need _int_mode as a member variable? Why not add it as an optional parameter?

File: src/modules/mavlink/mavlink_mission.cpp [hunk 1]
```diff
@@ -363,7 +397,10 @@ MavlinkMissionManager::send(const hrt_abstime now)
 		_transfer_in_progress = false;
 
 	} else if (_state == MAVLINK_WPM_STATE_GETLIST && hrt_elapsed_time(&_time_last_sent) > _retry_timeout) {
-		/* try to request item again after timeout */
+		/* try to request item again after timeout,
+		 * toggle int32 or float protocol variant to try both */
+		_int_mode = !_int_mode;
```
Comment (julianoes) [inline src/modules/mavlink/mavlink_mission.cpp]: @dagar I think we need `_int_mode` as a member variable for this.

Comment (dagar) [inline src/modules/mavlink/mavlink_mission.cpp]: Alright, that makes sense.

**Pull request merged**